algorithms/linfa-logistic/src/lib.rs

```diff
@@ -969,6 +977,17 @@ mod test {
         );
     }
 
+    #[test]
+    fn simple_example_3() {
```

Comment:
Can you also add a multinomial version of this test to see if multinomila logistic regression suffers from the same issue?